packages/validator/src/util/externalSignerClient.ts

```diff
@@ -104,13 +107,28 @@ type Web3SignerSerializedRequest = {
 /**
  * Return public keys from the server.
  */
-export async function externalSignerGetKeys(externalSignerUrl: string): Promise<string[]> {
+export async function externalSignerGetKeys(externalSignerUrl: string, logger?: MinimalLogger): Promise<string[]> {
   const res = await fetch(`${externalSignerUrl}/api/v1/eth2/publicKeys`, {
     method: "GET",
     headers: {"Content-Type": "application/json"},
   });
 
-  return handlerExternalSignerResponse<string[]>(res);
+  if (!res.ok) {
+    const errBody = await res.text();
+    throw Error(`External signer GetKeys: statusCode=${res.status}, errorBody=${errBody}`);
+  }
+
+  const resBody = await res.text();
+  const resContentType = res.headers?.get("Content-Type")?.split(";", 1)[0].trim().toLowerCase();
+
+  logger?.debug("External signer GetKeys response", {contentType: resContentType, body: resBody});
+
+  if(resContentType === "application/json") {
+    return JSON.parse(resBody) as string[];
+  }
+  else {
+    throw Error(`External signer GetKeys: content type ${resContentType} not supported`);
```

Comment:
This is not the content type but rather the media type, should also make sure that it's clear in the error that this is not a issue with the request but rather the response


see an example
https://github.com/ChainSafe/lodestar/blob/f6e45e8f1f88a5a1799da35f0c6d2414fac5f29f/packages/api/src/utils/client/response.ts#L21

Reply:
refactored the code according to the example